packages/discovery-react-components/src/components/DocumentPreview/components/ReactPdfViewer/ReactPdfViewer.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { SFC, useEffect } from 'react';
+import { Document, Page, pdfjs } from 'react-pdf';
+import PdfjsWorkerAsText from 'pdfjs-dist/build/pdf.worker.min.js';
+
+// pdfjs.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${pdfjs.version}/pdf.worker.js`;
+
+setupPdfjs();
+
+interface Props {
+  /**
+   * PDF file data as base64-encoded string
+   */
+  file: string;
+  /**
+   * Page number, starting at 1
+   */
+  page: number;
+  /**
+   * Zoom factor, where `1` is equal to 100%
+   */
+  scale: number;
+  /**
+   * Callback invoked with page count, once `file` has been parsed
+   */
+  setPageCount: (count: number) => void;
+  /**
+   * Check if document is loading
+   */
+  setLoading: (loading: boolean) => void;
+  /**
+   * Callback which is invoked with whether to enable/disable toolbar controls
+   */
+  setHideToolbarControls?: (disabled: boolean) => void;
+}
+
+const ReactPdfViewer: SFC<Props> = ({
+  file,
+  page,
+  scale,
+  setPageCount,
+  setLoading,
+  setHideToolbarControls
+}) => {
+  useEffect(() => {
+    if (setHideToolbarControls) {
+      setHideToolbarControls(false);
+    }
+  }, [setHideToolbarControls]);
+
+  return (
+    <div>
+      <Document
+        file={`data:application/pdf;base64,${file}`}
+        onLoadSuccess={({ numPages }: any) => setPageCount(numPages)}
```

Comment:
Try not to use `any`.

Reply:
I tried using `number`, but eslint was not happy.  I think there's some underlying type stuff going on with `numPages` and how it's being handled by `react-pdf` that does not like `number`.  I went with `any` for the time being to just get something working.

Reply:
I think once you install `@types/react-pdf`, you won't need to put a type here.